packages/data-format/index.js

```diff
@@ -0,0 +1,126 @@
+/* eslint-disable no-unused-vars */
+import Big from 'big.js'
+Big.RM = 0
+Big.NE = -19
+
+export const fromDripToCfx = number => {
+  try {
+    const cfx = new Big(number).div(1e18).toString()
+    return cfx
+  } catch (e) {
+    return number
+  }
+}
+
+export const fromCfxToDrip = number => {
+  try {
+    const cfx = new Big(number).times(1e18).toString()
+    return cfx
+  } catch (e) {
+    return number
+  }
+}
+
+export const formatDigit = (num, digit) => {
+  try {
+    const bNum = new Big(num)
+    const str = num + ''
+    const strWithoutZero = removeZero(str)
+    const strArr = strWithoutZero.split('.')
+    // no digit
+    if (!digit) return num
+    // no decimal or digit < integer length
+    else if (!strArr[1] || strArr[0].length >= digit)
+      return toThousands(strArr[0])
+    else if (strArr[0].length + strArr[1].length <= digit)
+      return toThousands(strWithoutZero)
+    else {
+      const decimal = strArr[1].substring(0, digit - strArr[0].length)
+      return toThousands(removeZero([strArr[0], decimal].join('.')))
+    }
+  } catch (e) {
+    return num
+  }
+}
+
+export const toThousands = (num, delimiter = ',', prevDelimiter = ',') => {
+  try {
+    const bNum = new Big(num)
+    let str = num + ''
+    return str
+      .replace(new RegExp(prevDelimiter, 'igm'), '')
+      .split('.')
+      .reduce((acc, cur, index) => {
+        if (index) {
+          return `${acc}.${cur}`
+        } else {
+          return cur.replace(/(\d{1,3})(?=(\d{3})+(?:$|\.))/g, `$1${delimiter}`)
+        }
+      }, '')
+  } catch (e) {
+    return num
+  }
+}
+
+export const removeZero = num => {
+  try {
+    const bNum = new Big(num)
+    let str = num + ''
+    let end = str.length - 1
+    const strArr = str.split('.')
+    if (!strArr[1]) return str
+    else if (strArr[1].length === 0) return str.substring(0, end)
```

Comment:
`!''` is `true`
so if `!strArr[1]` is false
`else if (strArr[1].length === 0) return str.substring(0, end)` will never run

what's more, `new Big('1000.').toString(10)` returns `'1000'`, there's no way `str.split('.')` be `''`

Reply:
and don't use `else if` if there's `return` in the first clause

Reply:
on another look, big.js already does the job
```js
function trimZero(numOrStr) {
  try {
    return new Big(numOrStr).toString(10)
  } catch (err) {
    return numOrStr
  }
}
```

Reply:
remove function removeZero and use big.js toString(10) instead